common/types/map.go

```diff
@@ -111,49 +112,89 @@ func (m *baseMap) ConvertToNative(typeDesc reflect.Type) (interface{}, error) {
 				StructValue: jsonMap}}, nil
 	}
 
+	// Unwrap pointers, but track their use.
+	isPtr := false
+	if typeDesc.Kind() == reflect.Ptr {
+		typeDesc = typeDesc.Elem()
+		isPtr = true
+	}
+
 	// Non-map conversion.
-	if typeDesc.Kind() != reflect.Map {
+	if typeDesc.Kind() != reflect.Map && typeDesc.Kind() != reflect.Struct {
```

Comment:
Is this redundant with the control flow from the switch at line 140?  If so, delete this.  If not, add a little comment explaining why we need to bail out here.

Reply:
Yep, thanks.